fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java

```diff
@@ -106,6 +109,49 @@
                   lao -> lao == null ? new ArrayList<com.github.dedis.student20_pop.model.event.Event>() :
                           lao.getRollCalls().values().stream().collect(Collectors.toList()));
 
+  private final LiveData<List<com.github.dedis.student20_pop.model.RollCall>> mLaoRollCalls = Transformations
+          .map(mCurrentLao,
+                  lao -> lao == null ? new ArrayList<com.github.dedis.student20_pop.model.RollCall>() :
+                          lao.getRollCalls().values().stream().collect(Collectors.toList()));
+
+  private final LiveData<List<com.github.dedis.student20_pop.model.RollCall>> mLaoAttendedRollCalls = Transformations
+          .map(mCurrentLao,
+                  lao -> lao == null ? new ArrayList<com.github.dedis.student20_pop.model.RollCall>() :
+                          lao.getRollCalls().values().stream().filter(rollcall->rollcall.getState()== EventState.CLOSED).filter(rollcall->attendedOrOrganized(lao, rollcall)).collect(Collectors.toList()));
+
+  /**
+   * Predicate used for filtering rollcalls to make sure that the user either attended the rollcall or was the organizer
+   * @param lao
+   * @param rollcall
+   * @return
+   */
+  private boolean attendedOrOrganized(Lao lao, RollCall rollcall){
+    //find out if user is the organizer
+    boolean isOrganizer = false;
+    try {
+      KeysetHandle publicKeysetHandle =
+              mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
+      isOrganizer =
+              lao.getOrganizer().equals(Keys.getEncodedKey(publicKeysetHandle));
+    } catch (GeneralSecurityException e) {
+      Log.d(TAG, "failed to get public keyset handle", e);
+      return false;
+    } catch (IOException e) {
+      Log.d(TAG, "failed to get public key", e);
+      return false;
+    }
```

Comment:
If the idea here is to know if the user attended or organised the roll calls of the current LAO then you can use the method `isOrganizer`